modules/openxr/config.py

```diff
@@ -1,10 +1,15 @@
 def can_build(env, platform):
-    if platform in ("linuxbsd", "windows", "android"):
-        return env["openxr"] and not env["disable_3d"]
-    else:
-        # not supported on these platforms
+    if not env["openxr"] or env["disable_3d"]:
         return False
 
+    if platform not in ("linuxbsd", "windows", "android"):
+        return False
+
+    if platform == "linuxbsd" and env["opengl3"] and not env["x11"]:  # Needs Xlib for OpenGL
+        return False
```

Comment:
I think this would cause the openxr module to not get built if both Vulkan and OpenGL are enabled, but X11 is disabled, right?

While that's still sort of non-sensical because X11 is the only way to render on Linux (until Wayland support is added), its not quite right semantically, because OpenXR should be able to build with Vulkan support, because it doesn't need to access any GLX internals when using Vulkan.